src/pages/index.js

```diff
@@ -45,45 +50,133 @@ const handleCardClick = (cardData) => {
   popupWithImage.open(cardData.link, cardData.name);
 };
 
-const createCard = (cardData) => {
-  const card = new Card(cardData, "#card-template", handleCardClick);
+const handleDeleteClick = (cardData, cardEl) => {
+  console.log("handleDeleteClick", deleteConfirmationPopup);
+  deleteConfirmationPopup.open();
+  console.log(cardEl);
+  deleteConfirmationPopup.setId(cardData._id);
+  deleteConfirmationPopup.setCard(cardEl);
+  console.log(deleteConfirmationPopup.getId());
+};
+
+const handleLikeCard = (card) => {
+  console.log(card);
+  api.toggleLikeOnCard(card._cardData._id, card._isLiked).then((response) => {
+    console.log(response);
+    card.updateLikes(response.likes);
+  });
+};
+
+const createCard = (cardData, userId) => {
+  const card = new Card(
+    cardData,
+    userId,
+    "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeCard
+  );
   return card.createCardElement();
 };
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      section.addItem(cardElement);
-    },
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co",
+  headers: {
+    authorization: "35337f3b-35e8-4dc0-a9b5-b6c4dd4127c3",
+    "Content-Type": "application/json",
   },
-  ".gallery__cards"
-);
+});
 
-section.renderItems();
+let cardList;
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  pictureSelector: ".profile__image",
 });
 
+let userId;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([user, cards]) => {
+    // userInfo.setProfileInfo(userData.name, userData.about);
+    userInfo.updateUserInfo({ name: user.name, job: user.about });
+    userInfo.updateProfileAvatar(user.avatar);
+    userId = user._id; // set user id
+
+    const section = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const cardElement = createCard(item, userId);
+          section.addItem(cardElement);
+        },
+      },
+      ".gallery__cards"
+    );
+
+    section.renderItems();
+    cardList = section;
+  })
+  .catch(() => {});
+
 const handleProfileFormSubmit = (e) => {
   e.preventDefault();
   // profileTitle.textContent = profileNameInput.value;
   // profileDescription.textContent = profileDescriptionInput.value;
   // editProfileFormValidator.disableSubmitButton();
   const inputValues = profileEditPopup.getInputValues();
-  userInfo.setUserInfo({
-    name: inputValues.Name,
-    job: inputValues.Title,
-  });
+  // userInfo.setUserInfo({
+  //   name: inputValues.Name,
+  //   job: inputValues.Title,
+  // });
+  toggleSaving("#profile-edit-modal");
+  api
+    .updateUserInfo({ name: inputValues.Name, about: inputValues.Title })
+    .then((result) => {
+      toggleSaving("#profile-edit-modal");
```

Comment:
Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally